### PR TITLE
Refine resource deletion protection proposal

### PR DESCRIPTION
**Pull request by XiShanYongYe-Chang**
**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:

Refine resource deletion protection proposal

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: docs/proposals/resource-deletion-protection/README.md [hunk 1]
```diff
@@ -19,43 +19,112 @@ creation-date: 2023-07-17
...
 
-#### CRD Example
+#### New label added
+
+```go
+    // ResourceDeletionProtectionLabelKey is the label key used to add to the resource to
+	// indicate the deletion protection policy of the resource. 
+	ResourceDeletionProtectionLabelKey = "resourcetemplate.karmada.io/deletion-protected"
+	// ResourceDeletionProtectionAlways indicate that deletion protection is always performed.
+	ResourceDeletionProtectionAlways = "Always"
```
Comment (RainbowMango) [inline docs/proposals/resource-deletion-protection/README.md:65]: indent

File: docs/proposals/resource-deletion-protection/README.md [hunk 2]
```diff
@@ -19,43 +19,112 @@ creation-date: 2023-07-17
...
 
-As a manager, I want to protect my `MySQL` application, so that they won't be accidentally deleted.
+As an end user, I want to protect my `MySQL` application, so that they won't be accidentally deleted.
 
 Our users may need to provide further protection for certain resources, not limited to just `Namespace`. They might want to secure resources such as `Deployment`,`StatefulSet`,`Service` and so on, for example, **all resources related to `MySQL` application**.
 
 ### Design Details
 
-We can add a CRD `ResourceDeletionProtection` to specify the resources which should be protected.
+Add a new label `resourcetemplate.karmada.io/deletion-protected` with enumerated values including `Always` and `Never`. Users can set this label for the resource to be protected. When the label value is set to `Always`, the user performs a delete operation, and `karmada-webhook` will reject the request, thus preventing the target resource form being deleted.
```
Comment (RainbowMango) [inline docs/proposals/resource-deletion-protection/README.md:56]: The enumerated values not include `Never`.

**Review by RainbowMango — COMMENTED** [review]
Generally looks good to me, just some nits.

File: docs/proposals/resource-deletion-protection/README.md (hunk 1 above)
Comment (XiShanYongYe-Chang) [reply docs/proposals/resource-deletion-protection/README.md]: Updated, thx~

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**